### ISSUE TITLE
My bug report

### ISSUE THREAD

**Issue by samuelzamvil** [PowerShell/PowerShell#13319]
## Steps to reproduce

Input:
```powershell
echo -n "test" | shasum -a 512
```

Output:
```
0e3e75234abc68f4378a86b3f4b32a198ba301845b0cd6e50106e874345700cc6663a86c1ea125dc5e92be17c98f9a0f85ca9d5f595db2012f7cc3571945c123  -
```

Input:
```powershell
"test" | shasum -a 512
```
Output: 
```
0e3e75234abc68f4378a86b3f4b32a198ba301845b0cd6e50106e874345700cc6663a86c1ea125dc5e92be17c98f9a0f85ca9d5f595db2012f7cc3571945c123  -
```

Input:
```powershell
echo -n 'test' | openssl dgst -sha512
```
Output:
```
0e3e75234abc68f4378a86b3f4b32a198ba301845b0cd6e50106e874345700cc6663a86c1ea125dc5e92be17c98f9a0f85ca9d5f595db2012f7cc3571945c123
```

Input:
```powershell
echo -n 'test' | md5
```

Output:
```
d8e8fca2dc0f896fd7cb4cb0031ba249
```

## Expected behavior

When using zsh or bash I receive the correct output
Input: "test" | shasum -a 512
Output: ee26b0dd4af7e749aa1a8ee3c10ae9923f618980772e473f8819a5d4940e0db27ac185f8a0e1d5f84f88bc887fd67b143732c304cc5fa9ad8e6f57f50028a8ff

Input: echo -n 'test' | md5
Output: 098f6bcd4621d373cade4e832627b4f6

## Actual behavior

Attempting to use built in binaries for calculating the hashes of a string fails on Mac OS. I'm not sure if I am doing something wrong or if there's a way to strip the object metadata(not sure if this is the right terminology) from the string and pass it to a linux binary.

Using: md5 -s 'test' does work correctly so I assume the issue only occurs when the string is passed through the pipeline.

## Environment data

Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Darwin 18.7.0 Darwin Kernel Version 18.7.0: Thu Jun 18 20:50:10 PDT 2020; root:xnu-4903.278.43~1/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

**Comment by rjmholt** [comment]
The results of these hash operations are going to depend heavily on encoding -- a string in either bash or PowerShell will have an implicit encoding and the "correct" behaviour depends on what you assume that encoding to be

**Comment by samuelzamvil** [comment]
> The results of these hash operations are going to depend heavily on encoding -- a string in either bash or PowerShell will have an implicit encoding and the "correct" behaviour depends on what you assume that encoding to be

Thank you for the response, it being an encoding issue makes a ton of sense. However, after a bit more research it turned out to be powershell appending a new line to the piped string. I confirmed that the sha/md5 sums that I was computing were matched what powershell computed if I omitted the `-n` in echo when using a different shell.

My google-fu failed until I added encoding to my search and it looks like the exact same issue was discussed here.
https://stackoverflow.com/questions/48371447/piping-text-to-an-external-program-appends-a-trailing-newline

Also there is an open issue here.
https://github.com/PowerShell/PowerShell/issues/1908
